shell/platform/android/platform_view_android.cc

```diff
@@ -26,22 +28,35 @@
 
 namespace flutter {
 
-std::unique_ptr<AndroidSurface> SurfaceFactory(
-    std::shared_ptr<AndroidContext> android_context,
+AndroidSurfaceFactory::AndroidSurfaceFactory(
+    std::shared_ptr<AndroidContext> context,
     std::shared_ptr<PlatformViewAndroidJNI> jni_facade) {
-  FML_CHECK(SurfaceFactory);
-  switch (android_context->RenderingApi()) {
+  android_context_ = context;
+  jni_facade_ = jni_facade;
+}
+
+AndroidSurfaceFactory::~AndroidSurfaceFactory() = default;
+
+void AndroidSurfaceFactory::SetExternalViewEmbedder(
+    std::shared_ptr<AndroidExternalViewEmbedder> external_view_embedder) {
+  external_view_embedder_ = external_view_embedder;
+}
+
+std::unique_ptr<AndroidSurface> AndroidSurfaceFactory::CreateSurface() {
+  FML_CHECK(external_view_embedder_);
```

Comment:
Since this is required now, what about passing `external_view_embedder` to the constructor of `AndroidSurfaceFactory`, and removing `SetExternalViewEmbedder`?

Reply:
External view embedder needs a reference to `AndroidSurfaceFactory`, that's the reason for having this method. I'm refactoring the relationship further in my coming commits, will simplify this.